### ISSUE TITLE
Memory Error Issue

### ISSUE THREAD

**Issue by vrajesh978** [lipim12/CS6140-ML-Doodle-Recognition#1]
```python
train_X,valid_X,train_ground,valid_ground = train_test_split(train_data,
                                                             train_data,
                                                             test_size=0.2,
                                                             random_state=13)
```

It gives memory Error issue.

I think we can use `keras generators` to solve this.

**Comment by lipim12** [comment]
train_test_split still needs to load the entire data in memory to shuffle. I doubt if the keras generator would help.
